examples/titanic.ipynb

```diff
@@ -0,0 +1,516 @@
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "-- let's compute the mean age. We can exclude\n",
+    "-- the null rows using the function `filterJust`.\n",
+    "-- (Similarly, you could exclude the nonnull rows\n",
+    "-- using the function `filterNothing`)\n",
+    "\n",
+    "\n",
+    "df |> D.filterJust \"Age\" |> D.mean (F.col @Double \"Age\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "id": "50baa76c-c3e3-4988-ab1e-051c2b58d270",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "  ----------------------------------------------------------------------------------------------  \n",
+       "| Age<br>Double | Survived<br>Int | isFemale<br>Int | isFirstClass<br>Int | isSecondClass<br>Int |\n",
+       "| --------------|-----------------|-----------------|---------------------|--------------------- |\n",
+       "| 22.0          | 0               | 0               | 0                   | 0                    |\n",
+       "| 38.0          | 1               | 1               | 1                   | 0                    |\n",
+       "| 26.0          | 1               | 1               | 0                   | 0                    |\n",
+       "| 35.0          | 1               | 1               | 1                   | 0                    |\n",
+       "| 35.0          | 0               | 0               | 0                   | 0                    |\n"
+      ],
+      "text/plain": [
+       "  ----------------------------------------------------------------------------------------------  \n",
+       "| Age<br>Double | Survived<br>Int | isFemale<br>Int | isFirstClass<br>Int | isSecondClass<br>Int |\n",
+       "| --------------|-----------------|-----------------|---------------------|--------------------- |\n",
+       "| 22.0          | 0               | 0               | 0                   | 0                    |\n",
+       "| 38.0          | 1               | 1               | 1                   | 0                    |\n",
+       "| 26.0          | 1               | 1               | 0                   | 0                    |\n",
+       "| 35.0          | 1               | 1               | 1                   | 0                    |\n",
+       "| 35.0          | 0               | 0               | 0                   | 0                    |"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "cleanedDf = df |> D.select [\"Sex\", \"Pclass\", \"Age\", \"Survived\"]\n",
+    "               |> D.derive \"isFemale\" (F.ifThenElse (F.col \"Sex\" .== \"female\") 1 0)\n",
+    "               |> D.derive \"isFirstClass\" (F.ifThenElse (F.col \"Pclass\" .== 1) 1 0)\n",
+    "               |> D.derive \"isSecondClass\" (F.ifThenElse (F.col \"Pclass\" .== 2) 1 0)\n",
+    "               |> D.impute (F.col \"Age\") 29.69911764705882\n",
```

Comment:
After seeing how clunky imputation was I added a function `imputeWith`.

Now you can write:

```haskell
imputeWith F.mean (F.col "Age")
```

And you can replace `F.mean` with any other aggregation function (median etc).